transform/variants/variants_test.go

```diff
@@ -1,39 +1,95 @@
-package variants_test
+package variants
 
 import (
 	"fmt"
-	"log"
 	"sort"
 	"testing"
 
-	"github.com/TimothyStiles/poly/transform/variants"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestIUPAC(t *testing.T) {
 	testSeq := "ATN"
 	testVariants := []string{"ATG", "ATA", "ATT", "ATC"}
-	testVariantsIUPAC, err := variants.AllVariantsIUPAC(testSeq)
-	if err != nil {
-		log.Fatal(err)
-	}
+	testVariantsIUPAC, err := AllVariantsIUPAC(testSeq)
+	assert.Nil(t, err)
 
 	sort.Strings(testVariants)
 	sort.Strings(testVariantsIUPAC)
 
 	for index := range testVariants {
-		if testVariants[index] != testVariantsIUPAC[index] {
-			t.Errorf("IUPAC variant has changed on test 'allIUPAC('ATN')'. Got this:\n%s instead of \n%s", testVariantsIUPAC, testVariants)
-		}
+		assert.Equal(t, testVariants[index], testVariantsIUPAC[index], "IUPAC variant has changed")
 	}
 }
 
+func TestIUPAC_errors(t *testing.T) {
+	testSeq := "ATX"
+	seqVariants, err := AllVariantsIUPAC(testSeq)
+	assert.NotNil(t, err, "expected error for unsupported IUPAC character, got nil")
+	assert.Equal(t, seqVariants, []string{})
+}
+
 func ExampleAllVariantsIUPAC() {
 	// AllVariantsIUPAC takes a string as input
 	// and returns all iupac variants as output
 	mendelIUPAC := "ATGGARAAYGAYGARCTN"
 	// ambiguous IUPAC codes for most of the sequences that code for the protein MENDEL
-	mendelIUPACvariants, _ := variants.AllVariantsIUPAC(mendelIUPAC)
+	mendelIUPACvariants, _ := AllVariantsIUPAC(mendelIUPAC)
 	fmt.Println(mendelIUPACvariants)
 	// Output: [ATGGAGAATGATGAGCTG ATGGAGAATGATGAGCTA ATGGAGAATGATGAGCTT ATGGAGAATGATGAGCTC ATGGAGAATGATGAACTG ATGGAGAATGATGAACTA ATGGAGAATGATGAACTT ATGGAGAATGATGAACTC ATGGAGAATGACGAGCTG ATGGAGAATGACGAGCTA ATGGAGAATGACGAGCTT ATGGAGAATGACGAGCTC ATGGAGAATGACGAACTG ATGGAGAATGACGAACTA ATGGAGAATGACGAACTT ATGGAGAATGACGAACTC ATGGAGAACGATGAGCTG ATGGAGAACGATGAGCTA ATGGAGAACGATGAGCTT ATGGAGAACGATGAGCTC ATGGAGAACGATGAACTG ATGGAGAACGATGAACTA ATGGAGAACGATGAACTT ATGGAGAACGATGAACTC ATGGAGAACGACGAGCTG ATGGAGAACGACGAGCTA ATGGAGAACGACGAGCTT ATGGAGAACGACGAGCTC ATGGAGAACGACGAACTG ATGGAGAACGACGAACTA ATGGAGAACGACGAACTT ATGGAGAACGACGAACTC ATGGAAAATGATGAGCTG ATGGAAAATGATGAGCTA ATGGAAAATGATGAGCTT ATGGAAAATGATGAGCTC ATGGAAAATGATGAACTG ATGGAAAATGATGAACTA ATGGAAAATGATGAACTT ATGGAAAATGATGAACTC ATGGAAAATGACGAGCTG ATGGAAAATGACGAGCTA ATGGAAAATGACGAGCTT ATGGAAAATGACGAGCTC ATGGAAAATGACGAACTG ATGGAAAATGACGAACTA ATGGAAAATGACGAACTT ATGGAAAATGACGAACTC ATGGAAAACGATGAGCTG ATGGAAAACGATGAGCTA ATGGAAAACGATGAGCTT ATGGAAAACGATGAGCTC ATGGAAAACGATGAACTG ATGGAAAACGATGAACTA ATGGAAAACGATGAACTT ATGGAAAACGATGAACTC ATGGAAAACGACGAGCTG ATGGAAAACGACGAGCTA ATGGAAAACGACGAGCTT ATGGAAAACGACGAGCTC ATGGAAAACGACGAACTG ATGGAAAACGACGAACTA ATGGAAAACGACGAACTT ATGGAAAACGACGAACTC]
+}
 
+func ExampleAllVariantsIUPAC_error() {
```

Comment:
I wasn't sure, but I thought it would be easier to add one in and remove it if it was deemed unnecessary, rather than add one in later if the reviewers felt we needed it?

Reply:
That's fair. My opinion is that it would be better to have test that checks for error instead of example, just because the example shows up in the generated docs, and we want to keep them clean.

Reply:
@fergloragain do you know if this will actually render to go docs? I'm not sure if it should be this way or not but I'm interested in seeing what it would look like.

Reply:
@fergloragain @Koeng101 Just checked. It does actually render to docs. MERGING.